ext/odbc/php_odbc.c

```diff
@@ -2604,14 +2629,14 @@ static void php_odbc_lasterror(INTERNAL_FUNCTION_PARAMETERS, int mode)
 	zval *pv_handle = NULL;
 	char *ret;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|r!", &pv_handle) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|O!", &pv_handle, odbc_connection_ce) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	if (pv_handle) {
-		if (!(conn = (odbc_connection *)zend_fetch_resource2(Z_RES_P(pv_handle), "ODBC-Link", le_conn, le_pconn))) {
-			RETURN_THROWS();
-		}
+		conn = Z_ODBC_CONNECTION_P(pv_handle);
+		CHECK_ODBC_CONNECTION(conn);
+
 		if (mode == 0) {
```

Comment:
This parameter name is really bad as I was confused where the hell the ``mode`` value came from, as I didn't see it from the userland call.